src/test/java/bio/terra/service/dataset/flight/create/DatasetCreateFlightTest.java

```diff
@@ -40,16 +42,26 @@ void testDatasetCreateFlightAuthorizeBillingSteps(boolean isTDRBillingProfile) {
     inputParameters.put(JobMapKeys.TDR_BILLING_PROFILE_FALLBACK.getKeyName(), isTDRBillingProfile);
     var flight = new DatasetCreateFlight(inputParameters, context);
 
+    List<String> stepNames = FlightTestUtils.getStepNames(flight);
+
     if (isTDRBillingProfile) {
       assertThat(
           "Dataset creation flight locates the billing info and then has two optional steps",
-          FlightTestUtils.getStepNames(flight),
+          stepNames,
           containsInRelativeOrder("AuthorizeBillingProfileUseStep"));
+      assertThat(
```

Comment:
To avoid the need for a negative test, you could replace these cases with the complete list of steps, as is done `SetInheritStewardFlightTest`.